nativeauthenticator/nativeauthenticator.py

```diff
@@ -295,13 +295,13 @@ def get_authed_users(self):
     def user_exists(self, username):
         return self.get_user(username) is not None
 
-    def create_user(self, username, pw, **kwargs):
+    def create_user(self, username, pw, pwc, **kwargs):
```

Comment:
I think we can make a new password confirmation check ahead of calling `create_user()`, it seems out of scope for such function to also validate the new password confirmation.

Reply:
I thought about this while writing the code and decided to have the confirmation logic here (instead of in the handlers) because this function also already contains the logic for other checks that can conceivably fail and could be handled elsewhere, like usernames being taken, being invalid and passwords not being strong enough. It seemed in line to put the confirmation here, too.

If we factor out the confirmation of passwords, which of the others should also be factored out?

Reply:
> I thought about this while writing the code and decided to have the confirmation logic here (instead of in the handlers)

But now you have it in both places, right? Because you have adjusted the message based on the confirmation being okay or not.

> If we factor out the confirmation of passwords, which of the others should also be factored out?

I'd say the username validation and password validation logic would make sense to extract, but perhaps not the username is taken logic as that would strictly make creating a user fail in a database perspective. I think of this as a scope-of-PR-creep where we would end up doing something unrelated to the PR to refactor that now though.

I think the validation of username/password is at least done on things that are otherwise still required to be passed to the create_user function, but logically, I think everything besides the bare minimum validations of what is required to actually add a userinfo entry to the database should be extracted.

Reply:
Let's stick with the current topic then and factor out the rest in a different PR. If I know what should and shouldn't be in there, that gives me enough to work with.